### ISSUE TITLE
border-zero: A value of `0` is not allowed. `0` must be used.

### ISSUE THREAD

**Issue by richarddewit** [sasstools/sass-lint#861]
**What version of Sass Lint are you using?**
1.9.1

**Please include any relevant parts of your configuration**
```
border-zero:
    - 1
    - convention: 0
```

What did you do? Please include the actual source code causing the issue.
```
.element {
  border: 0;
}
```

**What did you expect to happen?**
Not raising a warning, since the actual value is the same as the expected value.

**What actually happened? Please include any error messages given to you by Sass Lint.**
(Using Atom's plugin)
```
[sass-lint] [Warning] [border-zero] A value of `0` is not allowed. `0` must be used.
```

**If you're using a IDE plugin have you tried the CLI too?**
Same output:
```
143:22  warning  A value of `0` is not allowed. `0` must be used                   border-zero
```

**Comment by thijskramer** [comment]
I think this problem could be solved by changing this in [border-zero.js](https://github.com/sasstools/sass-lint/blob/develop/lib/rules/border-zero.js#L31):

**From:**
```js
if (node.content === '0' || node.content === 'none') {
  //...
}
```
**To:**
```js
if (node.content === 0 || node.content === 'none') {
  //...
}
```

Configurations with `convention: 0` won't generate false warnings this way.

**Comment by richarddewit** [comment]
As I can see now, the default is set to the string `'0'`, instead of the int. When using a string, it doesn't produce a warning.

I'll pass in a PR to fix this, so that the int 0 could also be used.

**Comment by richarddewit** [comment]
Btw, @thijskramer - your solution doesn't fix it, it just skips the validation. `node.content` is the value in `border: 0;` or `border: none;` which is always a string. Therefore it will not pass your if-statement because `node.content` is `'0'` (string), not `0` (int) or `'none'`

**Comment by thijskramer** [comment]
Allright, well then this issue could be fixed by parsing the yml config file in a different way, because people like me are using a `.sass-lint.yml` file with the following config:
```yml
rules:
  border-zero:
    - 1
    - convention: 0
```

OR, maybe use just two equal signs to compare (I'm not sure if this would work though): 
```js
if (node.content == '0' || node.content === 'none') {
  //...
}
```

Which solution do you prefer?

**Comment by richarddewit** [comment]
The one I've made the PR with :) (convert convention to string)

**Comment by DanPurdy** [comment]
Yeah i think @richarddewit's solution is the way to go, we've added a few comments there so if you'd like to update it then we can merge it in for you. If not we can take it further ourselves and get it sorted, with proper accreditation to your contribution of course! 🎉

**Comment by richarddewit** [comment]
Sorry for the late reaction, but I currently can't find time to implement this. If you could pick this up, that would be nice :)
